File: Servidor_java/Modelos/Capa.java
package Servidor_java.Modelos;

public class Capa extends Produto{
    private String cor;
    private String modelo;
    private String material;
    

    public Capa(int id,String nome,String descricao,double preco,int estoque,String cor,String modelo,String material){

        super(id,nome,descricao,preco,estoque);

        this.cor = cor;
        this.modelo = modelo;
        this.material = material;
    }

    public String getcor(){return cor;}
    public String getModelo(){return modelo;}
    public String getMaterial(){return material;}

    public void setcor(String cor){this.cor = cor;}
    public void setModelo(String modelo){this.modelo = modelo;}
    public void setMaterial(String material){this.material = material;}

    public String toString() {
        return "Capinha [id=" + getId() + ", nome=" + getNome() + ", preco=" + getPreco() + ", estoque=" + getEstoque() + "cor="+ cor + "modelo="+ modelo + "material="+ material +"]";
    }

}

File: Cliente_cpp/src/modelos/Capa.cpp
#include "modelos/Capa.h"
#include <string>
#include <iostream>

Capa::Capa(int id, const std::string& nome, const std::string& descricao, double preco, int quantidade_estoque, const std::string& material, const std::string& modelo)
            : Produto(id, nome, descricao, preco, quantidade_estoque), material(material), modelo(modelo) {}
TipoProduto Capa::getTipo() const {
    return TipoProduto::CAPA;
}

void Capa::mostrar() const {
    Produto::mostrar();
    std::cout << "Material: " << material << std::endl;
    std::cout << "Modelo: " << modelo << std::endl;
}


std::string Capa::getMaterial() const {
    return material;
}

void Capa::setMaterial(const std::string& material) {
    this->material = material;
}

std::string Capa::getModelo() const {
    return modelo;
}

void Capa::setModelo(const std::string& modelo) {
    this->modelo = modelo;
}
